aten/src/ATen/native/AutogradComposite.cpp

```diff
@@ -22,6 +22,54 @@ std::tuple<at::Tensor, at::Tensor> _unpack_dual(const at::Tensor& tensor, int64_
   return std::tuple<at::Tensor, at::Tensor>(tensor._fw_primal(level), tensor._fw_grad(level));
 }
 
+Tensor _new_zeros_with_same_feature_meta(
+    const at::Tensor& self,
+    const at::Tensor& other,
+    c10::optional<int64_t> self_num_batch_dims) {
+  // NB: This function can be called directly from _set_fw_grad or
+  //     if self is batched, from this function's batching rule.
+  auto other_sizes = other.sizes();
+  auto other_strides = other.strides();
+  auto other_storage_offset = other.storage_offset();
+  int64_t other_storage_numel = other.storage().nbytes() / other.itemsize();
+
+  if (!self_num_batch_dims.has_value()) {
+    auto new_tensor = at::zeros({other_storage_numel}, other.options());
+    return new_tensor.as_strided(other_sizes, other_strides, other_storage_offset);
+  }
+
+  int64_t _self_num_batch_dims = self_num_batch_dims.value();
+
+  auto self_sizes = self.sizes();
+  auto self_strides = self.strides();
+
+  // NB: We don't check that the sizes of self is the same as that of other
+  //     because this function is also used in the inplace over view case
+
+  std::vector<int64_t> out_sizes;
+  out_sizes.reserve(other.dim() + _self_num_batch_dims);
+  out_sizes.insert(out_sizes.end(), self_sizes.begin(), self_sizes.begin() + _self_num_batch_dims);
+  out_sizes.insert(out_sizes.end(), other_sizes.begin(), other_sizes.end());
+
+  // We use the strides of other, and tack on the strides computed with
+  // the batch dims of self, so that the slices are arranged contiguously
+  std::vector<int64_t> out_strides;
+  out_strides.reserve(other.dim() + _self_num_batch_dims);
+
+  int64_t prod = other_storage_numel;
+  for (size_t i = 0; i < _self_num_batch_dims; ++i) {
+    out_strides.insert(out_strides.begin(), prod);
+    prod *= self_strides[i];
```

Comment:
Are we sure we want self_strides? It might be better to just pick new strides to avoid weird cases. For example, if batch_size is [2, 3, 4] and the other_storage_numel is 10, then we could pick strides of size [3 * 4 * 10, 4 * 10, 10] instead of re-using whatever whacky strides are in `self_strides`.

Reply:
I think it is already doing what we want? Since the first `_self_num_batch_dims` strides in self_strides will the batch_sizes which we multiply with `other_storage_numel`.